homeassistant/components/homematic/light.py

```diff
@@ -77,6 +77,9 @@ def effect(self):
 
     def turn_on(self, **kwargs):
         """Turn the light on and/or change color or color effect settings."""
+        if ATTR_TRANSITION in kwargs:
+            self._hmdevice.setValue('RAMP_TIME', kwargs[ATTR_TRANSITION])
```

Comment:
That is all time available?

Reply:
Well, it is available on all homematic dimmer devices and the light service is alligned in pymatic to the respective dimmer devices. So should work in any case, but i cannot test with all devices, while own only a subset :-)

Reply:
Check the HomeMaticIP datapoints too. Otherwise, test first if this exists before we set it.

Reply:
`RAMP_TIME` is a valid parameter for HomeMatic IP devices as well. But just in case there are exceptions to this rule, the `setValue`-call could be wrapped in a try/except block where in case of an exception we log a warning that transitions are not supported for that specific entity.
Adding this won't break anything since it's optional in the service call and nobody has it yet. Those who want to give it a try will see the effect or not and then get the warning and can take the parameter out of the service call again.

Reply:
This doesn't work for my HMIP -BDT. If i use: 
```yaml
entity_id: light.essbereich_deckenleuchte
transition: 20.0
brightness: 255
```

I get following error: 
`HMGeneric.setValue: RAMP_TIME on 0008DA499C8DB2:1 Exception: <Fault -5: 'Invalid parameter or value'>
`

I see that RAMP_TIME is used on channel 1, but the RAMP_TIME parameter is on channel 4. 
![image](https://user-images.githubusercontent.com/12832850/74678438-436a5e00-51bb-11ea-8f4e-b3c9657fcc1c.png)

To turn on the light, a channel is used (self._channel). I think for the RAMP_TIME also the channel self._channel should be used.

```suggestion
            self._hmdevice.setValue('RAMP_TIME', kwargs[ATTR_TRANSITION], self._channel)
```

Correct me if i'm wrong.

Best regards
hasenbole

Reply:
That is handle pyhomematic

Reply:
Just tested this. The transition works for HMIP-BDT, if the self._channel parameter is added to the setValue Method.

Reply:
> That is handle pyhomematic

Do you mean, the correct channel selection should be handled by pyhomematic?

Reply:
> Just tested this. The transition works for HMIP-BDT, if the self._channel parameter is added to the setValue Method.

@trilu2000 Can you please check if this change does effect your homematic dimmer devices? Which dimmer do you use? Which channel does your device use for ramp_time? If the change does not effect your dimmer devices, i would open a PR.